tests/utils/test_list_gsoc.py

```diff
@@ -0,0 +1,124 @@
+import unittest
+
+import bs4
+
+from funderfinder.utils.list_gsoc import (
+    get_curr_year_project,
+    get_curr_year_projects,
+    get_early_archive_project,
+    get_link_matches,
+    get_modern_archive_projects,
+)
+
+from ..context import funderfinder
+
+
+class TestListGSOC(unittest.TestCase):
+    def test_get_link_match_org(self):
+        self.assertEqual(
+            ["EOL"],
+            get_link_matches(
+                """<p>The Encyclopedia of Life (EOL) is a free, online
+        collaborative encyclopedia intended to document all of the 1.9 million living speciesÂ known to science.
+        It is compiled from existing databases and from contributions by experts and non-experts
+        throughout the world.</p>
+        <p>EOL developers are spread between United States, United Kingdom, Egypt and Philippines. Organization's
+        code is located atÂ https://github.com/EOL</p>"""
+            ),
+        )
+
+    def test_get_link_match_repo(self):
+        self.assertEqual(
+            ["beamcommunity/beamcommunity.github.com"],
+            get_link_matches(
+                """https://github.com/beamcommunity/beamcommunity.github.com/wiki"""
+            ),
+        )
+
+    def test_get_link_match_gist(self):
+        # github gists should not match
+        self.assertEqual(
+            [],
+            get_link_matches(
+                "https://gist.github.com/sniok/fb6a1bcf4cf78303a874fa2389739764"
+            ),
+        )
+
+    def test_get_early_archive_project_ext_link(self):
+        fragment = """
+        <span class="mdl-list__item-primary-content">
+        <span class="small-logo-box mdl-list__item-icon">
+        <i class="material-icons small-logo">business</i>
+        </span>
+        <a href="/archive/gsoc/2015/orgs/beamcommunity">BEAM Community</a>
```

Comment:
Similar confusion to below comment here with the `2015` year.

Reply:
Same as above, fixed!